snapshot/storage/metastore.go

```diff
@@ -0,0 +1,69 @@
+package storage
+
+import (
+	"context"
+
+	"github.com/docker/containerd/snapshot"
+)
+
+// MetaStore is used to store metadata related to a snapshot driver. The
+// MetaStore is intended to store metadata related to name, state and
+// parentage. Using the MetaStore is not required to implement a snapshot
+// driver but can be used to handle the persistence and transactional
+// complexities of a driver implementation.
+type MetaStore interface {
+	// TransactionContext creates a new transaction context.
+	TransactionContext(ctx context.Context, writable bool) (context.Context, Transactor, error)
+
+	// Stat returns the snapshot stat Info directly from
+	// the metadata.
+	Stat(ctx context.Context, key string) (snapshot.Info, error)
+
+	// Walk iterates through all metadata for the stored
```

Comment:
I think it would be a good idea to be explicit here. Right now the answer is no, but I think doing batched reads and only calling outside of a transaction is probably a much safer route.